### PR TITLE
Migrate reference/roles to v2, change updated field format to iso8601

### PR DESCRIPTION
**Pull request by purple-waves**
Fixes an inconsistent datetime format bug in reference/roles. Left the existing endpoint untouched, copied most of its functionality to /v2/reference/roles and made the datetime formats consistent.

Testing:
For volunteers and supervisors, GET request should return 
```
[
    {
        "id": 1,
        "code": "basic",
        "name": "Basic",
        "created": "2021-04-10T10:03:32",
        "updated": "2021-04-10T10:03:32",
        "deleted": "False"
    },
    {
        "id": 2,
        "code": "driver",
        "name": "Driver",
        "created": "2021-04-10T10:03:32",
        "updated": "2021-04-10T10:03:32",
        "deleted": "False"
    },
    {
        "id": 3,
        "code": "crewLeader",
        "name": "Crew Leader",
        "created": "2021-04-10T10:03:32",
        "updated": "2021-04-10T10:03:32",
        "deleted": "False"
    },
    {
        "id": 4,
        "code": "advanced",
        "name": "Advanced",
        "created": "2021-04-10T10:03:32",
        "updated": "2021-04-10T10:03:32",
        "deleted": "False"
    }
]
```

For supervisors only, the POST method should return the id of the new role in the database, and the new role should be visible in the `roles` table. If the request is malformed then the method returns a `null` id and does not modify the database.
Example post body:
```
{
    "name": "Test Role",
    "code": "testRole"
}
```

Example response:
```
{
    "id": "8"
}
```
Volunteers are not allowed to use the patch, post or delete methods.

Unauthenticated users should not be able to access any of the methods.

### CODE REVIEW

**Review by Xinyun-Zhou — COMMENTED** [review]
Everything looks good, could add the supervisor's authority to test.

File: controllers/v2/reference/api.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from flask_restful import Resource, marshal_with, reqparse
+
+from controllers.v2.v2_blueprint import v2_api
+from domain import session_scope
+from controllers.v2.reference.response_models import role_model, id_model
+from repository.reference_repository import get_roles, add_role, toggle_role, \
+    delete_role
+from services.jwk import requires_auth
+
+role_parser = reqparse.RequestParser()
```
[the remark this answers is not in the document]
Comment (purple-waves) [reply controllers/v2/reference/api.py:10]: What refactoring are you thinking of?

**Comment by purple-waves** [comment]
Thanks for the comments. I've added the supervisor auth for post, patch, delete and added tests to postman for post.

File: controllers/v2/reference/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (purple-waves) [reply controllers/v2/reference/api.py:10]: I'll chuck some tests in for delete and patch in postman, but at this point in time only the get is being used in the frontend.

**Comment by 090997** [comment]
all looks good. there is no other question.

**Review by 090997 — APPROVED** [review]

**Review by Xinyun-Zhou — APPROVED** [review]
Good job! Please rebase before you merge it!

**Pull request merged**